docs/containers-storage-architecture.md

```diff
@@ -0,0 +1,76 @@
+## containers-storage 1 "May 2024"
+
+## NAME
+containers-storage-architecture - The implementation of containers-storage 
+
+NOTE: These are implementation details subject to change. This document
+is intended to be used only as an aid for debugging and development
+of the containers-storage system. Do *not* directly access the filesystem
+components below from external code, only from the Go API.
+
+## DESCRIPTION
+The *containers/storage* library manages the following object types:
+
+- layer: A filesystem tree, along with metadata
```

Comment:
… organized in, and optimized for, a tree of parent/child links, where a single “layer” may either serve as a complete filesystem tree, or a difference from the parent.